lib/Service/RecentlyEditedFilesSource.php

```diff
@@ -45,26 +46,61 @@ public function __construct(IServerContainer $serverContainer,
 		$this->l10n = $l10n;
 	}
 
+	/**
+	 * @return bool
+	 */
+	private function isNodeExcluded(Node $node, bool $showHidden): bool {
+		$next = $node;
+
+		while (!$showHidden) {
+			if ($next->getName()[0] == ".")
+				return true;
+
+			try {
+				$next = $next->getParent();
+			} catch (NotFoundException $e) {
+				break;
```

Comment:
won't you run into an infinite loop here?

I think you would be better off with recursion. If the current node is hidden, you return `false`. If there is a parent you return `$this->isNodeExcluded($node->getParent(), $showHidden);`.

Reply:
Thank you for taking a look!

The sequence ends when a dotfile is found (`return false`) or the root node is found (`break`). In other words, it is always bounded by the depth of the file system hierarchy. If there were a clearer way to identify the root node other than by exception handling then I would be more than happy to use it.

While I agree that recursion is generally a more natural approach to tree traversal problems, I don't think anything is lost in this context by iterating. In the recursive case, we would want to introduce a new function so that `isNodeExcluded` can be adapted to other types of exclusions related to the target node (regex, mimetype, tag, etc.).